### ISSUE TITLE
Hyper-parameters to reproduce DeltaNet results are in the paper but not in the repo

### ISSUE THREAD

**Issue by kazuki-irie** [automl/unlocking_state_tracking#3]
Hello,

First of all, thanks a lot to the authors for making this repository publicly available---I found this very helpful!

I'm leaving a note here in case it's useful to others interested in reproducing the (DeltaNet) results from the paper.

The paper’s appendix provides a clear set of hyperparameters (another big thanks to the authors), but these are not reflected in the example commands in the README. 

In my attempt, running the commands in the README file as-is did not reproduce the modular arithmetic results reported in the paper. *However*, using the hyperparameters listed in the appendix did work.

For instance, the appendix indicates that a successful modular arithmetic configuration for DeltaNet uses 3 layers (compared to 2 for the parity task), and a batch size of 1024. These are not mentioned in this repo.

I'm not submitting a pull request since I've been working with a different model implementation, but I thought it might still be useful to share this here in case somebody else is having a problem reproducing results.

PS: as a side note, in my experience with a model closely related to DeltaNet (but slightly different) implemented using `fla` (https://github.com/fla-org/flash-linear-attention). I found that setting `initializer_range` (an `fla` model parameter) to 0.006 (rather than the default 0.02) improved stability. Also, training for 20,000 steps (with batch size 1024) was sufficient to solve modular arithmetic (without brackets), as opposed to the 100,000 steps reported in the paper.

**Comment by roger-yt** [comment]
Hi! Thank you for sharing, it is a really helpful reminder. However, I found that setting the layers to 3 and bs to 1024 still does not reproduce the performance reported in the paper. Is it possible for you to share the specific yaml file you used to train DeltaNet [-1, 1]? Thank you!

**Comment by kazuki-irie** [comment]
As I did not use the DeltaNet[-1, 1] implementation from this repo (instead, I had my own model which contains DeltaNet as a component in each layer; in a 3-layer 4-head 128-dim model), I do not really have a yaml configuration that can be used as is.

The most I can share is the training and dataset sections from my final/best configuration (which deviates a bit from the authors' configurations in the end; AFAIR, I further tweaked the learning rate and batch size for my setting):

```
training:
  batch_size: 256
  lr: 0.001
  seed: 42
  val_every_step: 200
  lr_warmup_steps: 10000
  lr_decay_until_steps: ${.num_steps}
  lr_decay_factor: 1e-3
  min_lr: 0.000001
  weight_decay: 0.1
  compile: false
  grad_clip_norm: 1.0
  num_steps: 100000
  device: cuda
  amp_precision: bfloat16
  weight_precision: float32
  enable_mixed_precision: true

dataset:
  name: form_language
  kwargs:
    synth_lang_type: modular_arithmetic
    vocab_size: 10
    seed: 42
    enable_mask: true
    context_length: 256
    min_sequence_length: 3
    max_sequence_length: 40
    count:
      train: 102400000
      validation: 8192
      test: 8192
    subpar:
      validation:
        min_sequence_length: 40
        max_sequence_length: 256
      test:
        min_sequence_length: 40
        max_sequence_length: 256
```

Good luck!
